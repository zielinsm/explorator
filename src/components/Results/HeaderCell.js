import React from "react";
import { Table } from "semantic-ui-react";

import "./HeaderCell.css";

const HeaderCell = props => {
  const { label, key, value } = props.cell;
  const { cellClick } = props;

  const onClick = function onCellClick(e) {
    cellClick(e, value);
  };

  return (
    <Table.HeaderCell onClick={onClick} className="HeaderCell" key={key}>
      {label}
    </Table.HeaderCell>
  );
};

export default HeaderCell;
