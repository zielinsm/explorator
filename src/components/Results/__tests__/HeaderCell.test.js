import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import HeaderCell from "../HeaderCell";

const mockData = { label: "mockCell", key: 1, value: "mockValue" };

describe("Tab component:", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<HeaderCell cell={mockData} />);
    expect(wrapper.find(".HeaderCell")).toHaveLength(1);
  });

  it("handles clicks properly", () => {
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <HeaderCell cell={mockData} cellClick={clickHandler} />
    );
    wrapper.find(".HeaderCell").simulate("click");
    expect(clickHandler).toBeCalled();
  });

  it("snapshot matches", () => {
    const tree = renderer.create(<HeaderCell cell={mockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
