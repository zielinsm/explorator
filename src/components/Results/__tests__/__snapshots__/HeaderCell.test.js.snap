// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component: snapshot matches 1`] = `
<th
  className="HeaderCell"
  onClick={[Function]}
>
  mockCell
</th>
`;
